core/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobContainer.java

```diff
@@ -97,6 +96,7 @@ public InputStream readBlob(String name) throws IOException {
     @Override
     public void writeBlob(String blobName, InputStream inputStream, long blobSize) throws IOException {
         final Path file = path.resolve(blobName);
+        // TODO: why is this not specifying CREATE_NEW? Do we really need to be able to truncate existing files?
         try (OutputStream outputStream = Files.newOutputStream(file)) {
```

Comment:
Tests rely on this broken behavior, so i opened https://github.com/elastic/elasticsearch/issues/15579 for it. 

HDFS plugin acts correctly and passes the equivalent of CREATE_NEW